k9mail-library/src/test/java/com/fsck/k9/mail/filter/FixedLengthInputStreamTest.java

```diff
@@ -0,0 +1,230 @@
+                "mKxHxyUuggufOjgJGvUoUooMBpMXDGrNohNzJbQmnCuFVHxulXEircCYtezkVeEkJKFRDXUlwvOSrfhTmnoytjeQbBcIaRZYsRAn" +
+                "isNlMSHqEXQIDirVGicrcOUxQzYhZlnHqwbKslxGkAjIPesZeQKSgbDMnFaAmVLOOrONYuszgMCNCvNGYJFOWGpZUmNvWvTWFoEX" +
+                "QrPSvEmhxyYmFuIfJppvJSDlggOwcXYiXdtgoFfRZyTQbQmdnhoVHvQffkcMSqHDGRpwSLRbqDWwHJOkVJSMFzHdttJwyVpWuTDa" +
+                "uFXOuHfpoXkHStuKkreyWLsasXTRVchqRaEYjUmcTuNZYlcVPFezKqIaDiRcRxuuYFtMzJqOIwMNmITofhdQiXTIfUHGfsYnPYgM" +
+                "wpVdzGoyIrSSKJjWMnGdhmJHwHoFCqNtXOnbltOMnlCuzBsNUIlrKgfHRTAzAoFAlMvxgIrGffqCLtpAdbCbxzxMuivOctNDhuXy" +
+                "iTuSYUvHnxZxklJDuMbMeGqQddtmAMVAhGNPVfdBaNQiDNxrRqLqFuYMjkDwekzDSgkFbCEiHPwmAqAuoRDrtnwXQfkhyKLNlhRE" +
+                "ZxDybKsCrrQPcrNFKskrvceHPGjvyBtVgaHTSmtzdMSVwNyMIlmsMIgZqpVCNVBmwPVYkEfuenVdGeQcSdGSEMuJGOdWXyMbVQiN" +
+                "hKKwnoffMoKJEbfgSLOMsSVUwEBRGFBPFhhsKLZrmPIdSMKwxhPOvzGywVCamswfIspSUHXyvtQrWGNUkyVyYWwjwumRSVfDFepT" +
+                "JnNHLfbAdkCfHsEbUQCHBlSlgpYExOjYyuYbLjCjBYdNXcfdShRRGhzcdsxmsbAnHunqzIeVsOrlJRfdhaDYtmeIOUBCqFZBpLXT" +
+                "jkvAhXYTjLCmcKebgCUFWCLgYnfxYIrTnbLuEyNPeIlyNIWefQrpYKHtEaUVESWrHYFWGHObaKnpspDilNhkfrWdFXUyZEgkbBPC" +
+                "DSUAsUKPpFNyrWPZCPHzVARrspCJbXNCkotJGtgrDdLMMGIPLEPirRZzRxYgDiefAsUvCTnQXGHTOXTCRgjZCegPBxXTblJVFepV" +
+                "YzPbwggNJDRgrHbtLvEQilSMnHSmeJahMUjJXXHFwZXQlMRVrPolpogKooCqIYErEiNnaWrbOaQhLRVxGCGHVnSaHyqaFQwdaSUF" +
+                "JRjJWyDvRjwIAVYTEghKdtdHnATaBxrExrpBmIdbZciYLtZszlxmziScLwdJWaXVycSAdubKkZGlozHnjBVRqllIzFufoSLgxExE";
+
+        STR_LENGTH = string.length();
+    }
+
+    // Test the cases that fixed length is smaller than the length of bytes array
+
+    @Test
+    public void testLongInput() throws IOException {
+        fixedLengthInputStream = new FixedLengthInputStream(new ByteArrayInputStream(string.getBytes()), LENGTH_LIMIT1);
+
+        final int LENGTH = 10000;   // LENGTH > LENGTH_LIMIT1
+        byte[] bytes = new byte[LENGTH_LIMIT1];
+
+        Assert.assertEquals(fixedLengthInputStream.read(bytes, 0, LENGTH), LENGTH_LIMIT1);
+        Assert.assertArrayEquals(bytes, string.substring(0, LENGTH_LIMIT1).getBytes());
+    }
+
+    @Test
+    public void testLengthLimit() throws IOException {
+        fixedLengthInputStream = new FixedLengthInputStream(new ByteArrayInputStream(string.getBytes()), LENGTH_LIMIT1);
+
+        fixedLengthInputStream.read(new byte[LENGTH_LIMIT1], 0, LENGTH_LIMIT1);
+
+        int b = fixedLengthInputStream.read();
+        Assert.assertEquals(b, -1);
+    }
+
+    @Test
+    public void testRead() throws IOException {
+        fixedLengthInputStream = new FixedLengthInputStream(new ByteArrayInputStream(string.getBytes()), LENGTH_LIMIT1);
+
+        final int INDEX = 888;
+        final int ARRAY_LEN = 10000;
+
+        fixedLengthInputStream.read(new byte[ARRAY_LEN], 0, INDEX);
```

Comment:
Again, this doesn't guarantee INDEX bytes are read. My recommendation is to use okio to reliably skip X bytes.